dandi/download.py

```diff
@@ -706,15 +707,29 @@ def _download_file(
 
     # TODO: how do we discover the total size????
     # TODO: do not do it in-place, but rather into some "hidden" file
+    downloaded = 0
     for attempt in range(3):
         try:
-            downloaded = 0
+            if downloaded:
+                downloaded = max(downloaded - 2 * MAX_CHUNK_SIZE, 0)
             if digester:
                 downloaded_digest = digester()  # start empty
             warned = False
             # I wonder if we could make writing async with downloader
-            with open(path, "wb") as writer:
-                for block in downloader():
+            with open(path, "a+b") as writer:
+                first = True
+                for block in downloader(start_at=downloaded):
+                    if first:
+                        writer.seek(downloaded)
+                        if downloaded:
+                            blob1 = writer.read(len(block))
+                            if blob1 == block:
+                                downloaded += len(block)
+                                continue
+                            else:
+                                downloaded = 0
+                                break
```

Comment:
please add `.debug` level lgr messages in both cases

Reply:
Done.